javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/config/ResourceProviderPropertiesCustomizer.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling.config;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.sdk.autoconfigure.spi.AutoConfigurationCustomizer;
+import io.opentelemetry.sdk.autoconfigure.spi.AutoConfigurationCustomizerProvider;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+@AutoService(AutoConfigurationCustomizerProvider.class)
+public class ResourceProviderPropertiesCustomizer implements AutoConfigurationCustomizerProvider {
+
+  private static final Map<String, String> DISABLED_BY_DEFAULT_RESOURCE_PROVIDERS = new HashMap<>();
+
+  static {
+    DISABLED_BY_DEFAULT_RESOURCE_PROVIDERS.put(
+        "io.opentelemetry.contrib.aws.resource.BeanstalkResourceProvider", "aws");
+    DISABLED_BY_DEFAULT_RESOURCE_PROVIDERS.put(
+        "io.opentelemetry.contrib.aws.resource.Ec2ResourceProvider", "aws");
+    DISABLED_BY_DEFAULT_RESOURCE_PROVIDERS.put(
+        "io.opentelemetry.contrib.aws.resource.EcsResourceProvider", "aws");
+    DISABLED_BY_DEFAULT_RESOURCE_PROVIDERS.put(
+        "io.opentelemetry.contrib.aws.resource.EksResourceProvider", "aws");
+    DISABLED_BY_DEFAULT_RESOURCE_PROVIDERS.put(
+        "io.opentelemetry.contrib.aws.resource.LambdaResourceProvider", "aws");
+    DISABLED_BY_DEFAULT_RESOURCE_PROVIDERS.put(
+        "io.opentelemetry.contrib.gcp.resource.GCPResourceProvider", "aws");
+    // for testing
+    DISABLED_BY_DEFAULT_RESOURCE_PROVIDERS.put(
+        "io.opentelemetry.javaagent.tooling.config.ResourceProviderPropertiesCustomizerTest$Provider",
+        "test");
+  }
+
+  static final String DISABLED_KEY = "otel.java.disabled.resource.providers";
+  static final String ENABLED_KEY = "otel.java.enabled.resource.providers";
+
+  @Override
+  public void customize(AutoConfigurationCustomizer autoConfigurationCustomizer) {
+    autoConfigurationCustomizer.addPropertiesCustomizer(this::customize);
+  }
+
+  // VisibleForTesting
+  Map<String, String> customize(ConfigProperties config) {
+    Set<String> enabledProviders = new HashSet<>(config.getList(ENABLED_KEY));
+
+    List<String> enabled = new ArrayList<>();
+    List<String> disabled = new ArrayList<>();
+
+    for (Map.Entry<String, String> providerEntry :
+        DISABLED_BY_DEFAULT_RESOURCE_PROVIDERS.entrySet()) {
+      String providerName = providerEntry.getKey();
+      String providerGroup = providerEntry.getValue();
+      Boolean explictEnabled =
+          config.getBoolean(String.format("otel.instrumentation.%s.enabled", providerGroup));
```

Comment:
I'm not sure about using `otel.instrumentation` here as that flag is used to enable/disable instrumentations, but we are dealing with resource providers here.

Reply:
I think @trask brought up this idea that resource providers are a form of instrumentation.

Maybe `otel.instrumentation.resource-provider.%s.enabled`?

Reply:
Names are hard, I think it would be best to raise this in the SIG.

Reply:
SIG proposed using `otel.resource.providers.aws.enabled`

Reply:
changed @laurit